dot/parachain/availability-distribution/availability_distribution.go

```
@@ -157,10 +160,31 @@ func (ad *AvailabilityDistribution) handleChunkFetchingRequest(
 	return response, nil
 }
 
-//nolint:unused
 func (ad *AvailabilityDistribution) handlePoVFetchingRequest(
-	who peer.ID,
+	_ peer.ID,
 	payload []byte,
 ) (network.ResponseMessage, error) {
-	return nil, nil // TODO: implement
+	request := &messages.PoVFetchingRequest{}
+
+	err := request.Decode(payload)
+	if err != nil {
+		return nil, fmt.Errorf("decoding chunk fetching request: %w", err)
+	}
+
+	query := availabilitystore.QueryAvailableData{
+		CandidateHash: request.CandidateHash,
+		Sender:        make(chan availabilitystore.AvailableData),
+	}
+
+	ad.subSystemToOverseer <- query
+	response := &messages.PoVFetchingResponse{}
+
+	availableData := <-query.Sender
+	if availableData.PoV.BlockData == nil {
+		_ = response.SetValue(parachaintypes.NoSuchPoV{})
+	} else {
+		_ = response.SetValue(availableData.PoV)
+	}
```

Why do we ignore errors?
Can we please add error checks?

---

> Why do we ignore errors?

The parameters passed to `SetValue()` as well as `response` are static values that are known to work together. But you are right, better safe than sorry.